### PR TITLE
Add the option of selecting cells having any point satisfying a condition, not just all points

### PR DESCRIPTION
**Pull request by mohamed82008**
So that not all points have to satisfy the condition function, can also select cells with any point having some feature.

### CODE REVIEW

File: src/Grid/grid.jl [hunk 1]
```diff
@@ -120,15 +120,16 @@ function addcellset!(grid::Grid, name::String, cellid::Union{Set{Int}, Vector{In
     grid
 end
 
-function addcellset!(grid::Grid, name::String, f::Function)
+function addcellset!(grid::Grid, name::String, f::Function; all = true)
```
Comment (fredrikekre) [inline src/Grid/grid.jl]: `all::Bool=true`

File: src/Grid/grid.jl [hunk 2]
```diff
@@ -120,15 +120,16 @@ function addcellset!(grid::Grid, name::String, cellid::Union{Set{Int}, Vector{In
...
     cells = Set{Int}()
     for (i, cell) in enumerate(getcells(grid))
-        all_true = true
+        pass = all
         for node_idx in cell.nodes
             node = grid.nodes[node_idx]
-            !f(node.x) && (all_true = false; break)
+            all && !f(node.x) && (pass = false; break)
```
Comment (fredrikekre) [inline src/Grid/grid.jl]: Perhaps store `f(node.x)` in a variable instead of calling it twice?

Comment (mohamed82008) [reply src/Grid/grid.jl]: Sure will do it tomorrow, thanks.

Comment (mohamed82008) [reply src/Grid/grid.jl]: But won't it be called only once per iteration? Because `all && ...` and `!all && ...` are like if statements.

Comment (mohamed82008) [reply src/Grid/grid.jl]: Actually storing `f(node.x)` is still useful to avoid calling it multiple times for different cells sharing the same node.

Comment (fredrikekre) [reply src/Grid/grid.jl]: Right, this should be free anyway so doesn't really matter.

**Comment by fredrikekre** [comment]
I included this commit in #174, thanks!

**Pull request closed without merging**